chainstate/test-suite/src/tests/pos_processing_tests.rs

```diff
@@ -523,12 +523,29 @@ fn pos_block_signature(#[case] seed: Seed) {
     );
 
     // bad block signature
+    // Note: this tx is not strictly needed for the test, but without it the block id here
```

Comment:
Ususally we solve this problem with a different timestamp.

Reply:
I think this should never happen because the time is advanced after the first block. Previous block was invalid so unless we saved invalid block index the hash cannot be equal.

@ImplOfAnImpl do you have the seed of the test failure? I can take a look

Reply:
> I think this should never happen because the time is advanced after the first block.

Not sure I understand you correctly. In the test, both blocks re-use the same block_timestamp.

> Previous block was invalid so unless we saved invalid block index the hash cannot be equal.

We now do save indices of invalid blocks, that's exactly the reason for the test failure.

> @ImplOfAnImpl do you have the seed of the test failure? I can take a look

It's reproducible with any seed in my branch.

I'd say that the problem here is that several tests are packed into one test function and because of that they began interfering with each other. Perhaps I should split the test into several smaller tests?

Reply:
ok I see that we save these blocks, nevermind then